packages/form-state-manager/src/tests/utils/manager-api.test.js

```diff
@@ -1126,4 +1126,71 @@ describe('managerApi', () => {
       });
     });
   });
+
+  describe('initialize', () => {
+    it('reinitilize form', () => {
+      const render = jest.fn();
+
+      const managerApi = createManagerApi({});
+
+      managerApi().registerField({ name: 'foo', render });
+      managerApi().registerField({ name: 'bar', render });
+
+      managerApi().change('foo', 'foo');
+      managerApi().change('bar', 'baz');
+
+      expect(managerApi().pristine).toEqual(false);
+      expect(managerApi().values).toEqual({
+        foo: 'foo',
+        bar: 'baz'
+      });
+      expect(managerApi().initialValues).toEqual({});
+
+      managerApi().initialize({ foo: { bar: 'foobar' } });
+
+      expect(managerApi().pristine).toEqual(true);
```

Comment:
Can you add a check that the field.meta.prisinte/dirty/touched/visited were reset as well?